pkg/providers/vault/vault.go

```diff
@@ -61,6 +62,8 @@ func New(cfg api.StaticConfig) *provider {
 	if p.AuthMethod == "" {
 		if os.Getenv("VAULT_AUTH_METHOD") == "approle" {
 			p.AuthMethod = "approle"
+		} else if os.Getenv("VAULT_AUTH_METHOD") == "kubernetes" {
```

Comment:
If this isn't a standard environment variable defined by Vault, would you mind adding some documentation about using VAULT_AUTH_METHOD when you'd want to use k8s auth? Thanks!